main.py

```diff
@@ -82,35 +84,30 @@ def cmd_upload(args):
         print("INFO: uploading image")
         upload_image_from_url(glance, new_img.id, args.url)
 
-        # remove self from the list of tenants with which to share
-        if args.tenant_id in args.share_with:
-            args.share_with.remove(args.tenant_id)
+        # remove self from the list of projects with which to share
+        if args.project_id in args.share_with:
+            args.share_with.remove(args.project_id)
 
         if len(args.share_with) > 0:
-            print("INFO: sharing with tenants")
-            for tenant in args.share_with:
-                glance.image_members.create(new_img.id, tenant)
+            print("INFO: sharing with projects")
+            for project in args.share_with:
+                glance.image_members.create(new_img.id, project)
 
             # We actually don't *have* to do the below. The image is
-            # already accessible in the tenants, though they won't
+            # already accessible in the projects, though they won't
             # show up in a full listing until accepted. So let's try
             # to be nice.
 
-            # We only support sharing with tenants under the same
+            # We only support sharing with projects under the same
             # username/password.
 
             # XXX: Consider making this part optional since it assumes
-            # we have access to the other tenants.
-            accept_image_in_tenants(new_img.id, args.share_with,
-                                    args.project_domain_id, args.auth_url,
+            # we have access to the other projects.
+            accept_image_in_projects(new_img.id, args.share_with, args.auth_url,
                                     args.username, args.password)
 
-        if args.name:
-            # The Python API is not fully compatible with v2 yet,
-            # e.g. update() is spotty, so let's use v1.
-            glance1 = glance_session_from_args(args, 1)
-            print("INFO: renaming to", args.name)
-            glance1.images.update(new_img.id, name=args.name)
+        print("INFO: renaming to", args.name)
+        glance.images.update(new_img.id, name=args.name)
```

Comment:
Why drop the `if args.name`, though? The name is optional.

Reply:
good catch!  i might have deleted it when i was testing out the removal of the v1 stuff